File: js/src/formulas/diff.ts
export default function diff (f: (x: number) => number, density = 5) {
  const dx = 2 * (10 ** (-density))
  const dy = (x: number) => f(x + (10 ** (-density))) - f(x - (10 ** (-density)))

  return (x: number) => dy(x) / dx
}

File: cs/src/formulas/diff.cs
using System;

public static class Diff
{
    public static Func<double, double> Create(Func<double, double> f, int density = 5)
    {
        double dx = 2 * Math.Pow(10, -density);
        return x => (f(x + Math.Pow(10, -density)) - f(x - Math.Pow(10, -density))) / dx;
    }
}
